lib/ansible/playbook/role/__init__.py

```diff
@@ -288,7 +289,7 @@ def _prepend_validation_task(self, task_data):
         :returns: The (possibly modified) task list.
         '''
         if self._metadata.argument_specs:
-            if self._dependencies:
+            if C.config.get_config_value('VALIDATE_ROLE_WITH_DEPS_WARNING') and self._dependencies:
                 display.warning("Dependent roles will run before roles with argument specs even if validation fails.")
```

Comment:
I ran this by some Ansible community folks and they understood the implications based on this warning.
```suggestion
                display.warning("Role dependencies will run before the dependent role even if argument validation fails "
                                "for the dependent role. To disable this warning, set VALIDATE_ROLE_WITH_DEPS_WARNING to False.")
```